pkgs/top-level/splice.nix

```diff
@@ -170,5 +170,5 @@ in
 
   # Haskell package sets need this because they reimplement their own
   # `newScope`.
-  __splicedPackages = splicedPackages // { recurseForDerivations = false; };
+  __splicedPackages = if actuallySplice then splicedPackages // { recurseForDerivations = false; } else pkgs;
```

Comment:
Todo try early cutoff with splicedPackagesWithXorg

Reply:
Run 1 with that
![image](https://github.com/user-attachments/assets/be0d1c3a-3fbe-49a8-8c9b-a8c53fe288cb)